buttonhub.py

```diff
@@ -261,6 +271,80 @@ def do_mqtt(req, context, _client):
     _client.publish(topic, payload)
 
 
+def do_flow(req, context):
+    flow_name = req.get('flow', '')
+    flow = config.get('flows', {}).get(flow_name)
+    if not flow:
+        print("No Flow defined")
+        return ''
+    do_action(flow, flow_name, context)
+
+
+def schedule_flow(req, context):
+    global scheduled_flows
+    flow_name = req.get('schedule-flow', '')
+    schedule_time = None
+    if 'interval' in req:
+        timedelta = _parse_time_interval(req['interval'])
+        schedule_time = datetime.datetime.now() + timedelta
+    if not schedule_time:
+        return
+
+    scheduled_flows.append({
+        'name': flow_name,
+        'time': schedule_time,
+        'context': context,
+    })
+
+
+def _parse_time_interval(interval_string):
+    parts = interval_string.split(':')
+    hours = 0
+    minutes = 0
+    if len(parts) == 2:
+        hours = int(parts[0])
+        minutes = int(parts[1])
+    elif len(parts) == 1:
+        minutes = int(parts[0])
+    return datetime.timedelta(hours=hours, minutes=minutes)
+
+
+def cancel_scheduled_flow(req, context):
+    global scheduled_flows
+    flow_name = req.get('cancel-scheduled-flow', '')
+    scheduled_flows = [entry for entry in scheduled_flows if entry['name'] != flow_name]
```

Comment:
if multiple of the same flow are scheduled, this will cancel all. Is this intended, or deemed acceptable?